File: apps/services/auth-manager/src/main/java/com/nyad/authmanager/service/AuditService.java
package com.nyad.authmanager.service;

import com.nyad.authmanager.dto.audit.AuditAction;

public interface AuditService {
  void logAudit(String userId, AuditAction action, Class<?> entityClass, String entityId);

  void logAudit(String userId, AuditAction action, String entityName, String entityId);
}

File: apps/services/event-manager/internal/service/audit_service.go
package service

import (
	"eventsphere/event-manager/internal/client"
	"eventsphere/event-manager/internal/dto"
	"eventsphere/event-manager/internal/model/data"
)

type AuditService struct {
	Client *client.AuditClient
}

func (s *AuditService) LogAudit(userID string, action data.AuditAction, entityName string, entityID string) {
	req := dto.AuditLogRequestDTO{
		Action:     action,
		EntityID:   entityID,
		EntityName: entityName,
	}
	s.Client.CreateAudit(nil, req) // ctx is nil if called from background/exception
}